File: Salrunde/Room.m
//
//  Room.m
//  Salrunde
//
//  Created by Jan Matias Ørstavik on 23/01/15.
//  Copyright (c) 2015 Jan Matias Ørstavik. All rights reserved.
//

#import "Room.h"

@implementation Room

+(Room *)roomWithName:(NSString *)name Building: (NSString *)building ID: (NSString *)ID A4:(BOOL)A4 A3:(BOOL)A3 black:(BOOL)black color:(BOOL)color cardReader:(BOOL)cardReader xero:(BOOL)xero fuser:(BOOL)fuser
{
	Room *r = [[Room alloc] init];
	r.name = name;
	r.building = building;
	r.ID = ID;
	r.A4 = A4;
	r.A3 = A3;
	r.black = black;
	r.color = color;
	r.cardReader = cardReader;
	r.xero = xero;
	r.fuser = fuser;
	r.useLLF = NO;
	
	r.storage = NO;
	r.tynnklient = NO;
	return r;
}

+(Room *)roomWithName: (NSString *)name Building: (NSString *)building Lat: (NSString *)latitude Long: (NSString *)longitude Floor: (NSString *)floor A4:(BOOL) A4 A3:(BOOL) A3 black: (BOOL) black color: (BOOL) color cardReader: (BOOL) cardReader xero:(BOOL)xero fuser:(BOOL)fuser
{
	Room *r = [[Room alloc] init];
	r.name = name;
	r.building = building;
	r.latitude = latitude;
	r.longitude = longitude;
	r.floor = floor;
	r.A4 = A4;
	r.A3 = A3;
	r.black = black;
	r.color = color;
	r.cardReader = cardReader;
	r.xero = xero;
	r.fuser = fuser;
	r.useLLF = YES;
	
	r.storage = NO;
	r.tynnklient = NO;
	return r;
}

+(Room *)storageWithName:(NSString *)name Building: (NSString *)building ID: (NSString *)ID A4:(BOOL) A4 A3:(BOOL) A3 black: (BOOL) black color: (BOOL) color
{
	Room *r = [[Room alloc] init];
	r.name = name;
	r.building = building;
	r.ID = ID;
	r.A4 = A4;
	r.A3 = A3;
	r.black = black;
	r.color = color;
	r.cardReader = NO;
	
	r.storage = YES;
	r.tynnklient = NO;
	return r;
}

+(Room *)storageWithName: (NSString *)name Building: (NSString *)building Lat: (NSString *)latitude Long: (NSString *)longitude Floor: (NSString *)floor A4:(BOOL) A4 A3:(BOOL) A3 black: (BOOL) black color: (BOOL) color
{
	Room *r = [[Room alloc] init];
	r.name = name;
	r.building = building;
	r.latitude = latitude;
	r.longitude = longitude;
	r.floor = floor;
	r.A4 = A4;
	r.A3 = A3;
	r.black = black;
	r.color = color;
	r.useLLF = YES;
	
	r.storage = YES;
	r.tynnklient = NO;
	return r;
}

+(Room *)tynnklienterWithName:(NSString *)name Building: (NSString *)building
{
	Room *r = [[Room alloc] init];
	r.name = name;
	r.building = building;
	r.A4 = NO;
	r.A3 = NO;
	r.black = NO;
	r.color = NO;
	r.cardReader = NO;
	
	r.storage = NO;
	r.tynnklient = YES;
	return r;}

@end

File: Salrunde/Room.swift
//
//  Room.swift
//  Salrunde
//
//  Created by Jan Matias Ørstavik on 04/02/15.
//  Copyright (c) 2015 Jan Matias Ørstavik. All rights reserved.
//

import Foundation

class Room {
	
	// Boolean declarations
	let A4, A3, black, color, xero, fuser, cardReader : Bool
	let printerRoom, StorageRoom, computerRoom, activationClients : Bool
	let useCoordinatesForLocation : Bool
	
	// String declarations
	let name, building, ID, latitude, longitude, floor : String
	// Static initializers
	init(name : String, building : String, ID : String,  A4 : Bool, A3: Bool, black : Bool, color : Bool, xero : Bool, fuser : Bool, cardReader : Bool, printerRoom : Bool, StorageRoom : Bool, computerRoom : Bool, activationClients : Bool, useCoordinatesForLocation : Bool, latitude : String, longitude : String, floor : String){

		self.name = name
		self.building = building
		self.ID = ID
		self.A4 = A4
		self.A3 = A3
		self.black = black
		self.color = color
		self.xero = xero
		self.fuser = fuser
		self.cardReader = cardReader
		self.printerRoom = printerRoom
		self.StorageRoom = StorageRoom
		self.computerRoom = computerRoom
		self.activationClients = activationClients
		self.useCoordinatesForLocation = useCoordinatesForLocation
		self.latitude = latitude
		self.longitude = longitude
		self.floor = floor
	}
	
	
	
	
}